core/utils.go

```diff
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/spf13/viper"
```

Comment:
I think we should avoid using this package in MinBFT core. It seems to be too specific for a particular target of integration. Maybe we should communicate logging parameters via optional parameters of the replica constructor?

Reply:
@sergefdrv You're right, thanks. I'll update that part.

Reply:
Optional parameters could be passed as `opts ...Option` to `New()`. Then `Option` can be defined as `type Option func(*options)`, where `options` defined as `type options struct { /* some fields */ }`. Those parameters can be applied in `New()` like:
```go
	for _, opt := range opts {
		opt(&s.opts)
	}
```
And a concrete option can be implemented like:
```go
func WithLogFile(f *os.File) Option {
	return func(opts *options) {
		opts.logFile = f
	}
}
```
This way of implementing would be in line with what we already have prepared internally for view change implementation. Somewhat similar approach one can find used here https://godoc.org/github.com/micro/go-micro/client#NewStream